### PR TITLE
Uplift exporter code and repository

### PR DESCRIPTION
**Pull request by TobiasKadelka**
- Makes code flake8 compliant
- specifies imports (no `import *`)
- improves error handling
- adds `pyproject.toml`
- adds github workflow file

Updated version of previous PR https://github.com/psyinfra/prometheus-eaton-ups-exporter/pull/26

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: git config
+      run: |
+        git config --global user.email "eaton@exporter.com"
+        git config --global user.name "Eaton Exporter"
+    - name: Install dependencies
```
Comment (aqw) [inline .github/workflows/tests.yml:26]: Ideally these are defined in the `pyproject.toml` file. Such as:
```
[project.optional-dependencies]
tests = [
    'flake8',
    'pyre-check',
    'pytest',
    'pytest-cov',
    'pytest-randomly'
]
```

And then that target is installed.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      run: |
+        pip install . flake8
```
Comment (aqw) [inline .github/workflows/tests.yml]: IIRC your previous PR included Pyre. Any reason to drop it?

File: prometheus_eaton_ups_exporter/scraper.py [hunk 3]
```diff
@@ -3,11 +3,30 @@
 
 import urllib3
 from requests import Session, Response
-from requests.exceptions import SSLError, ConnectionError,\
-    ReadTimeout, MissingSchema, InvalidURL
-
+from requests.exceptions import (
+        SSLError,
```
Comment (aqw) [inline prometheus_eaton_ups_exporter/scraper.py]: Nit: alphabetization is easier to skim.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+keywords = ["prometheus", "exporter", "eaton"]
+license = {text = "ISC"}
+classifiers = [
+    "Programming Language :: Python :: 3",
+]
+dependencies = [
+    "prometheus_client",
+    "requests",
+    "urllib3",
+    "vcrpy",
```
Comment (aqw) [inline pyproject.toml]: This is only used for tests and should not be part of the normal dependencies.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+]
+
+[project.urls]
+"Bug Reports" = "https://github.com/psyinfra/prometheus-eaton-ups-exporter/issues"
+"Source" = "https://github.com/psyinfra/prometheus-eaton-ups-exporter/"
+
+[project.optional-dependencies]
+tests = [
+    "flake8",
+    "pyre-check",
```
Comment (aqw) [inline pyproject.toml]: Pyre is installed here but not used in the GitHub Actions workflow.

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+description = "A Prometheus exporter for Eaton UPSs."
+version = "1.2.0"
+readme = "README.md"
+requires-python = ">=3.11"
+keywords = ["prometheus", "exporter", "eaton"]
+license = {text = "ISC"}
+classifiers = [
+    "Programming Language :: Python :: 3",
+]
+dependencies = [
```
Comment (aqw) [inline pyproject.toml:19]: `requirements.txt` and `test-requirements.txt` should be folded into this file and removed.

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    "prometheus_client",
+    "requests",
+    "urllib3",
```
Comment (aqw) [inline pyproject.toml]: Out of scope, but this dependency is used in one place, and simply to disable SSL verification.

`requests` supports this feature directly. https://requests.readthedocs.io/en/latest/user/advanced/#ssl-cert-verification

**Review by aqw — CHANGES_REQUESTED** [review]
Looking good! `requirements.txt` et al are the only blockers.

Other things are nice to fix, but are up to you whether to do them in this PR or another.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (TobiasKadelka) [reply .github/workflows/tests.yml]: Yes, that it is complaining a lot about `Undefined import [21]: Could not find a module corresponding to import "vcr".` and many others.

And while I plan to look into this and fix it, I did not want that to be a blocker for the rest of the uplifting which IMO provides a nice ground for such fixes/addons.
Note: Another thing I want to check too is the pytests, which fail for me locally, and might be possible to add to the CI, too (at least in so far these can be run without and actual response).

Comment (aqw) [reply .github/workflows/tests.yml]: That happens for modules that don't have typing available. `ruamel` has the same issue.

You can pyre-ignore the import:
```
import vcr  # pyre-ignore[21] 
```

Comment (aqw) [reply .github/workflows/tests.yml]: > Note: Another thing I want to check too is the pytests, which fail for me locally, and might be possible to add to the CI, too (at least in so far these can be run without and actual response).

Ahh, that is indeed a problem. ;-)

File: pyproject.toml (hunk 7 above)
Comment (TobiasKadelka) [reply pyproject.toml]: I created an issue based on this, and will look into that in a future PR.

**Comment by TobiasKadelka** [comment]
pyre, pytest, and the removal of the dependency will be addressed at a later point.

I added changes for the rest of your feedback, especially in regards of the requirements!

**Comment by mathisloevenich** [comment]
Nice to see you'all keeping this project updated 👍

**Review by aqw — APPROVED** [review]
LGTM!

**Comment by aqw** [comment]
> Nice to see you'all keeping this project updated 👍

Absolutely!

Thanks for building an exporter with good bones. It's definitely worth maintaining! :-D

**Comment by mathisloevenich** [comment]
> > Nice to see you'all keeping this project updated 👍
> 
> Absolutely!
> 
> Thanks for building an exporter with good bones. It's definitely worth maintaining! :-D

Thanks, I appreciate that :)

File: pyproject.toml (hunk 7 above)
Comment (TobiasKadelka) [reply pyproject.toml]: I looked into this, and I think your comment is at least in parts wrong; for more details I commented in issue #28 to discuss this :)

**Pull request merged**